src/matrix/Client.ts

```diff
@@ -123,24 +140,31 @@ export class Client {
         return result;
     }
 
-    queryLogin(homeserver) {
+    queryLogin(homeserver: string): AbortableOperation<Promise<LoginOptions>, void> {
         return new AbortableOperation(async setAbortable => {
             homeserver = await lookupHomeserver(homeserver, (url, options) => {
                 return setAbortable(this._platform.request(url, options));
             });
             const hsApi = new HomeServerApi({homeserver, request: this._platform.request});
-            const response = await setAbortable(hsApi.getLoginFlows()).response();
+            const response = await (setAbortable(await hsApi.getLoginFlows()) as IHomeServerRequest).response();
```

Comment:
I think I get what you're trying to do with `setAbortable`'s typing, however typescript doesn't want to work that way. `SetAbortableFn` looks like:

```typescript
export type SetAbortableFn = (a: IAbortable) => typeof a;
```

which typescript interprets to mean 

```typescript
export type SetAbortableFn = (a: IAbortable) => IAbortable;
```

even though what you're really trying to say is more like

```typescript
export type SetAbortableFn = (a: IAbortable) => extends IAbortable;
```

I don't know if there's a way to express this in typescript without using generics, however then you're limited to passing one type of object into `SetAbortableFn` when really you want to be able to pass multiple different `IAbortable`s.